lib/pbench/test/unit/server/test_datasets_inventory.py

```diff
@@ -126,8 +124,7 @@ def mock_send_file(path_or_file, *args, **kwargs):
         response = query_get_as("fio_2", "f1.json", HTTPStatus.OK)
         assert response.status_code == HTTPStatus.OK
 
-        file_content, args = mock_args
+        _, args = mock_args
 
-        assert str(file_content).startswith("<_io.BytesIO")
```

Comment:
I don't think that removing the assertion at (old) line 131 is the best course.  Yeah, we shouldn't be doing the partial string compare, but why wouldn't something like this work?
```python
        assert isinstance(file_content, io.BytesIO)
```
(If `io.BytesIO` isn't the right type, then you can go further up the inheritance, say `io.BufferedIOBase` or `io.IOBase`.)

If you want to follow [Dave's suggestion](https://github.com/distributed-system-analysis/pbench/pull/3105#discussion_r1237601970), then, when you create `f1.json`, put something recognizable in it, and then read it here, like
```python
        assert file_content.read() == "<f1.json content>"
```

But, I [still think](https://github.com/distributed-system-analysis/pbench/pull/3105#discussion_r1237587969) you have the wrong "unit" here.

Reply:
Trying to read the content of byte stream which has been closed is throwing the error.

```
assert file_content.read() == "<f1.json content>"
E       ValueError: I/O operation on closed file.
```

Reply:
Ah; yeah, in theory you can `stream.seek(0)` to rewind it and read it again. I did that in the agent `results.py` so I could make one pass to compute the SHA256 and another to actually transfer the data. (Unfortunately, even exiting the `with` to close the file and then re-opening it didn't work, presumably because of some artifact of lazy Python object management, though there's probably some way to force it ... ultimately the rewind seemed easier.)

Reply:
> Trying to read the content of byte stream which has been closed is throwing the error.

Right...it would, of course.  😛

So, I think it would be simpler to test just the `_get()` function as the unit.  The existing test supplies mocks for `CacheManager.find_dataset()`, `Path.is_file()`, and Flask `send_file()`; if instead you supply mocks for `CacheManager.filestream()`, `send_file()`, and `close()`, then you get full control over the stream (i.e., you can have the `filestream()` mock return it and then the other two mocks can check to make sure that they get it if and when they are supposed to).  I think this would be much simpler than trying to observe and capture what the deeper levels are doing.